library-aux/src/main/scala/scala/runtime/Statics.scala

```diff
@@ -34,56 +34,35 @@ object Statics {
   }
 
   def longHash(lv: Long): Int = {
-    lv.toInt
-    /*
-    val iv = lv.toInt | 0
+    val iv = lv.toInt
     if (iv == lv) iv
-    else (lv ^ (lv >>> 32)).toInt | 0
-    */
+    else iv ^ (lv >>> 32).toInt
```

Comment:
In fact I realized there's a better implementation of this method, that does not involve allocating a new `RuntimeLong`:
```scala
val lo = lv.toInt
val hi = (lv >>> 32).toInt
if (hi == (lo >> 31)) lo
else lo ^ hi
```